diff -ur vmware-distrib.orig/bin/vm-support vmware-distrib/bin/vm-support
--- vmware-distrib.orig/bin/vm-support	2007-05-02 06:52:50.000000000 +0200
+++ vmware-distrib/bin/vm-support	2007-05-11 15:02:07.000000000 +0200
@@ -422,7 +422,7 @@
 runcmd "echo vm-support version: $VER" "/tmp/vm-support-version.$$.txt"
 runcmd "lspci -H1 -M" "/tmp/lspci1.$$.txt"
 runcmd "lspci -H1 -M -vn" "/tmp/lspci2.$$.txt"
-runcmd "/sbin/lsmod" "/tmp/modules.$$.txt"
+runcmd "/bin/lsmod" "/tmp/modules.$$.txt"
 runcmd "uname -a" "/tmp/uname.$$.txt"
 runcmd "df" "/tmp/df.$$.txt"
 runcmd "cat /etc/issue" "/tmp/issue.$$.txt"
diff -ur vmware-distrib.orig/bin/vmware-config.pl vmware-distrib/bin/vmware-config.pl
--- vmware-distrib.orig/bin/vmware-config.pl	2007-05-02 06:52:50.000000000 +0200
+++ vmware-distrib/bin/vmware-config.pl	2007-05-13 21:51:47.000000000 +0200
@@ -10121,7 +10121,7 @@
          error('Please shut down any running VMs and run this script again.' .
                "\n\n");
       } else {
-         if (system(shell_string(db_get_answer('INITSCRIPTSDIR') . '/vmware')
+         if (system(shell_string('/etc/rc.d' . '/vmware')
                     . ' stop')) {
             error('Unable to stop services for ' . vmware_product_name() .
                   "\n\n");
@@ -10155,13 +10155,13 @@
     configure_gtk2();
     createMimePackageFile();
     configureDesktopFiles();
-    configure_mon();
-    if (isDesktopProduct()) {
-      build_vmblock();
-    }
-    configure_pp();
+#    configure_mon();
+#    if (isDesktopProduct()) {
+#      build_vmblock();
+#    }
+#    configure_pp();
     configure_net();
-    build_vmnet();
+#    build_vmnet();
   }
 
    configure_eclipse_plugin();
@@ -10249,7 +10249,7 @@
   db_save();
   # Then start VMware's services
   if (!$gOption{'skip-stop-start'}) {
-    system(shell_string(db_get_answer('INITSCRIPTSDIR') . '/vmware') . ' start');
+    system(shell_string('/etc/rc.d' . '/vmware') . ' start');
     print "\n";
   }
 
diff -ur vmware-distrib.orig/bin/vmware-uninstall.pl vmware-distrib/bin/vmware-uninstall.pl
--- vmware-distrib.orig/bin/vmware-uninstall.pl	2007-05-02 06:52:33.000000000 +0200
+++ vmware-distrib/bin/vmware-uninstall.pl	2007-05-11 15:24:26.000000000 +0200
@@ -2876,7 +2876,7 @@
   }
 
   if (defined($gDBAnswer{'INITSCRIPTSDIR'})
-      && db_file_in(db_get_answer('INITSCRIPTSDIR') . $service_name)) {
+      && file_name_exist(db_get_answer('INITSCRIPTSDIR') . $service_name)) {
     # The installation process ran far enough to create the startup script
     my $status;
 
@@ -2893,7 +2893,7 @@
     }
 
     # Stop the services
-    $status = system(shell_string(db_get_answer('INITSCRIPTSDIR')
+    $status = system(shell_string('/etc/rc.d'
                                   . $service_name) . ' stop') >> 8;
     if ($status) {
       if ($status == 2) {
diff -ur vmware-distrib.orig/installer/services.sh vmware-distrib/installer/services.sh
--- vmware-distrib.orig/installer/services.sh	2007-05-02 06:52:51.000000000 +0200
+++ vmware-distrib/installer/services.sh	2007-05-11 14:20:20.000000000 +0200
@@ -608,14 +608,14 @@
 #
 vmmonUseCount() {
    # Beware of module dependancies here. An exact match is important
-   /sbin/lsmod | awk 'BEGIN {n = 0;} {if ($1 == "'"$driver"'") n = $3;} END {print n;}'
+   /bin/lsmod | awk 'BEGIN {n = 0;} {if ($1 == "'"$driver"'") n = $3;} END {print n;}'
 }
 
 # Is a given module loaded?
 isLoaded() {
    local module="$1"
 
-   /sbin/lsmod | awk 'BEGIN {n = "no";} {if ($1 == "'"$module"'") n = "yes";} END {print n;}'
+   /bin/lsmod | awk 'BEGIN {n = "no";} {if ($1 == "'"$module"'") n = "yes";} END {print n;}'
 }
 
 # Build a Linux kernel integer version
@@ -655,7 +655,7 @@
 }
 
 vmware_load_module() {
-   /sbin/insmod -s -f "/lib/modules/`uname -r`/misc/$1.o" || exit 1
+   /sbin/insmod -s -f "/lib/modules/`uname -r`/misc/$1.ko" || exit 1
    exit 0
 }
 
diff -ur vmware-distrib.orig/lib/net-services.sh vmware-distrib/lib/net-services.sh
--- vmware-distrib.orig/lib/net-services.sh	2007-05-02 06:52:28.000000000 +0200
+++ vmware-distrib/lib/net-services.sh	2007-05-11 14:54:38.000000000 +0200
@@ -582,7 +582,7 @@
 isLoaded() {
    local module="$1"
 
-   /sbin/lsmod | awk 'BEGIN {n = "no";} {if ($1 == "'"$module"'") n = "yes";} END {print n;}'
+   /bin/lsmod | awk 'BEGIN {n = "no";} {if ($1 == "'"$module"'") n = "yes";} END {print n;}'
 }
 
 # Check if there is an IP route for a given subnet via a given interface
