--- Makefile.org	2004-06-25 22:33:56.000000000 +0400
+++ Makefile	2006-10-09 19:48:43.000000000 +0400
@@ -4,7 +4,9 @@
 include ../platform-settings
 
 CFLAGS:=$(CFLAGS) -I../jabberd  `glib-config --cflags`
-LIBS:=$(LIBS) `glib-config --libs glib`
+LIBS:=$(LIBS) `glib-config --libs glib` -ljabberd -shared
+#CFLAGS:=$(CFLAGS) -I../jabberd -I../jabberd/lib -I/usr/lib/glib-2.0/include -I/usr/include/glib-2.0
+#LIBS:=$(LIBS) `glib-config --libs glib` -L../xdb_file/.libs -ljabberdxdbfile -L../jabberd/.libs -ljabberd
 
 YAHOO_OBJECTS=yahoo-transport.o yahoo-session.o yahoo-phandler.o yahoo.o \
 	crypt.o gaim-sha.o yahoo-presence.o yahoo-server.o md5.o yahoo-message.o \
