diff -wbBur sxe/sxeedit.cpp /home/sergej/sergej-repo.svn/psi-plus/src/psi-build/src/sxe/sxeedit.cpp
--- sxe/sxeedit.cpp	2010-12-13 09:27:44.000000000 +0000
+++ /home/sergej/sergej-repo.svn/psi-plus/src/psi-build/src/sxe/sxeedit.cpp	2010-12-14 12:48:32.000000000 +0000
@@ -69,7 +69,6 @@
 
 	// Can't compare edits to different records
 	if(rid() != other.rid())  {
-		qDebug() << QString("Comparing SxeEdits to %1 an %2.").arg(rid()).arg(other.rid()).toAscii();
 		return false;
 	}
 
diff -wbBur sxe/sxerecord.cpp /home/sergej/sergej-repo.svn/psi-plus/src/psi-build/src/sxe/sxerecord.cpp
--- sxe/sxerecord.cpp	2010-12-13 09:27:44.000000000 +0000
+++ /home/sergej/sergej-repo.svn/psi-plus/src/psi-build/src/sxe/sxerecord.cpp	2010-12-14 12:59:00.000000000 +0000
@@ -51,7 +51,6 @@
 		else if (edit->type() == SxeEdit::Record)
 			applySxeRecordEdit(dynamic_cast<SxeRecordEdit*>(edit));
 	} else {
-		qDebug() << QString("Tried to apply an edit meant for %1 to %2.").arg(edit->rid()).arg(rid()).toAscii();
 	}
 }
 
@@ -65,7 +64,6 @@
 
 bool SxeRecord::applySxeNewEdit(QDomDocument &doc, SxeNewEdit* edit) {
 	if(!(edits_.size() == 0 && node_.isNull())) {
-		qDebug("Someone's not behaving! Tried to apply a SxeNewEdit to an existing node.");
 		emit nodeRemovalRequired(node_);
 		return false;
 	}
@@ -114,13 +112,10 @@
 }
 
 void SxeRecord::revertToZero() {
+	const SxeNewEdit* edit = dynamic_cast<const SxeNewEdit*>(edits_[0]);
 #ifndef NDEBUG
 	if (edits_[0]->type() != SxeEdit::New)
-		qDebug() << QString("First edit is of type %1!").arg(edits_[0]->type());
 #endif
-
-	const SxeNewEdit* edit = dynamic_cast<const SxeNewEdit*>(edits_[0]);
-
 	parent_ = edit->parent();
 	primaryWeight_ = edit->primaryWeight();
 	identifier_ = edit->name();
@@ -216,16 +211,6 @@
 					data_.replace(from, n, edit->value(key));
 
 				} else {
-					if(!ok1)
-						qDebug() << QString("Could not convert 'replacefrom' = '%1' to int.").arg(edit->value(SxeRecordEdit::ReplaceFrom));
-					if(!ok2)
-						qDebug() << QString("Could not convert 'replacen' = '%1' to int.").arg(edit->value(SxeRecordEdit::ReplaceN));
-					if(from < 0)
-						qDebug() << QString("'replacefrom' = '%1' is negative.").arg(edit->value(SxeRecordEdit::ReplaceFrom));
-					if(n < 0)
-						qDebug() << QString("'replacen' = '%1' is negative.").arg(edit->value(SxeRecordEdit::ReplaceN));
-					if(from + n > data_.length())
-						qDebug() << QString("from (%1) + n (%2) > data_.length() (%3).").arg(from).arg(n).arg(data_.length());
 				}
 
 			} else {
@@ -257,7 +242,6 @@
 				edit->nullify();  // There's no way the edit could be applied anymore
 
 		} else {
-			qDebug() << QString("Edit of type %1 at %2!").arg(edits_[i]->type()).arg(i).toAscii();
 		}
 
 	}
@@ -288,7 +272,6 @@
 
 	if(data_ != lastData_) {
 
-		// qDebug() << QString("Setting '%1' to \"%2\"").arg(node_.nodeName()).arg(data_).toAscii();
 
 		if((node_.isText() || node_.isAttr() || node_.isComment())) {
 
